OpenRA.Mods.Common/Traits/BotModules/UnitBuilderBotModule.cs

```diff
@@ -22,8 +22,9 @@ public class UnitBuilderBotModuleInfo : ConditionalTraitInfo
 		// TODO: Investigate whether this might the (or at least one) reason why bots occasionally get into a state of doing nothing.
 		// Reason: If this is less than SquadSize, the bot might get stuck between not producing more units due to this,
 		// but also not creating squads since there aren't enough idle units.
-		[Desc("Only produce units as long as there are less than this amount of units idling inside the base.")]
-		public readonly int IdleBaseUnitsMaximum = 12;
+		[Desc("If > 0, only produce units as long as there are less than this amount of units idling inside the base.",
+			"Beware: if it is less than squad size, e.g. the `SquadSize` from `SquadManagerBotModule`, the bot might get stuck as there aren't enough idle units to create squad.")]
+		public readonly int IdleBaseUnitsMaximum = -1;
```

Comment:
Well, I think developers would prefer limit is disabled by default or it will be confusing when squad is stuck on not producing units.

> Though that usecase doesn't really even work. Units just get assigned to a squad and despite idling are no longer considered as idling

This can be solved with add a limit of the max number of each type squad in bot hand.

Reply:
I kinda think that this variable should either be removed in its entirety, or the calculation of idle units changed to include idle squad units in base

Reply:
Or the best idea is, change the design: SquadManager request units to build, UnitBuilder take the order to build.

If SquadManager find only hover units can reach the target, then it will build mostly hover unit on offensive. When SquadManager find it impossible to reach the target, it will just keep a protection squad and no order to UnitBuildet.